### ISSUE TITLE
Press Ctrl + left mouse button to drag the video screen to any position.

### ISSUE THREAD

**Issue by zhouxinghong** [mpv-player/mpv#15020]
### Expected behavior of the wanted feature

Press Ctrl + left mouse button`(Ctrl+MBTN_LEFT)` to drag the video screen to any position.

**Operation steps and function description:**

1. Change video zoom.
    **`Ctrl+Wheel up/down`**
2. Please support pressing Ctrl + left mouse button`(Ctrl+MBTN_LEFT)` to drag the video screen to any position.
    **`Ctrl+MBTN_LEFT`**
3. After the video screen is scaled, trigger **`Ctrl+MBTN_MID`** to reset the scaled video screen size to the default video screen size.

Windown 10
MPV 0.39

**Comment by zhouxinghong** [comment]
Example effect:
Similar to the magnifying glass effect of the video screen.

**Comment by zhouxinghong** [comment]
Example effect 4:

https://github.com/user-attachments/assets/a3bd02af-2507-42b0-a3b7-5c2a80b40d9e

**Comment by zhouxinghong** [comment]
Example effect 4:

https://github.com/user-attachments/assets/e3fece51-1536-4265-8582-b162481de4c3

**Comment by zhouxinghong** [comment]
2. Please support pressing Ctrl + left mouse button(Ctrl+MBTN_LEFT) to drag the video screen to any position.
Ctrl+MBTN_LEFT

Hello! Regarding the 2nd function, the range size of the video movement by dragging with the mouse is not enough, is it possible to customize the range size of the video movement by dragging.
Like the effect below.

Example effect 4:

https://github.com/user-attachments/assets/e3fece51-1536-4265-8582-b162481de4c3

**Comment by Trudeau566** [comment]
Change the drag-to-pan part to try
`mp.add_key_binding(nil, "drag-to-pan", function (t)
    if t.event == "up" then
        mp.remove_key_binding("drag-to-pan-mouse-move")
        return
    end

    local dims = mp.get_property_native("osd-dimensions")
    local old_mouse_pos = mp.get_property_native("mouse-pos")
    local old_pan_x = mp.get_property_number("video-pan-x") or 0.5
    local old_pan_y = mp.get_property_number("video-pan-y") or 0.5

    mp.add_forced_key_binding("MOUSE_MOVE", "drag-to-pan-mouse-move", function ()
        local mouse_pos = mp.get_property_native("mouse-pos")
        local dx = (mouse_pos.x - old_mouse_pos.x) / (dims.w * 0.7)  -- 水平移动系数，数字越小越快，越大越慢
        local dy = (mouse_pos.y - old_mouse_pos.y) / (dims.h * 0.7)  -- 垂直移动系数
        
        -- 修改方向：现在向下拖动会使画面向下移动（增加pan-y值）
        mp.set_property("video-pan-x", old_pan_x + dx)  -- 改为+
        mp.set_property("video-pan-y", old_pan_y + dy)  -- 改为+
    end)
end, { complex = true })`

**Comment by zhouxinghong** [comment]
Hello friend!
Can you please provide a command item that can reset (Ctrl+MBTN_LEFT) the position of the video after dragging and dropping to the original position of the video, the main purpose is for binding other mouse buttons or keyboard.

**Comment by guidocella** [comment]
Alt+Backspace resets it.

**Comment by zhouxinghong** [comment]
@guidocella Can I customize the bindings to other mouse buttons or keyboards?

**Comment by guidocella** [comment]
You can rebind it like any other key in input.conf. Just check what Alt+BS is bound to in the key binding menu or in the builtin input.conf.

**Comment by zhouxinghong** [comment]
@guidocella 
I can't seem to find this Alt+BS keyboard binding in input.conf!

**Comment by guidocella** [comment]
You can use the key binding menu or stats page 4.

**Comment by zhouxinghong** [comment]
> You can use the key binding menu or stats page 4.

I don't understand what you mean?
I can't find this keyboard binding in mpv.io Keyboard Control either.

**Comment by guidocella** [comment]
`g-b` shows the key binding menu and Shift+i and 4 the stats key binding list (https://mpv.io/manual/master/#stats)

**Comment by zhouxinghong** [comment]
@guidocella Is the following keybinding syntax correct and valid?
KP5 set video-zoom 0 ; set panscan 0 ; set video-pan-x 0 ; set video-pan-y 0 ; set video-scale-x "1" ; set video-scale-y "1"

**Comment by guidocella** [comment]
Yeah but that won't undo `drag-to-pan`. You need to set `video-align-x` and `video-align-y` to 0.

**Comment by zhouxinghong** [comment]
@guidocella 
Thank you very much for your careful help, sir! 🍦🏆👍👏😜

Also, is this panning feature something that could be reconsidered to further improve it?

**Comment by guidocella** [comment]
No, going beyond the window is annoying and a second completely different implementation using `video-pan` would have to be shipped to do it. You can use the version in https://github.com/occivink/mpv-image-viewer.

**Comment by zhouxinghong** [comment]
But in most cases for a close look at a video in vertical screen (shot by a cell phone), it is really necessary to pan beyond the window to view the details in the video more accurately, so I hope you can consider this feature request as it is a useful feature.

**Comment by zhouxinghong** [comment]
There is also an issue that the process is not smooth and fluid when using the mouse `drag-to-pan` video.

Problems to be solved:

Problem 1: The process is not smooth and fluid when using mouse `drag-to-pan` video.

**Comment by guidocella** [comment]
Dunno it is fluid for me.

**Comment by zhouxinghong** [comment]
> Dunno it is fluid for me.

I have tried in QMPlay2 that the process when using mouse `drag-to-pan` video is very smooth, very smooth and very natural. You can try it too.

**Comment by guidocella** [comment]
QMPlay2 is more fluid but mpv's version is also plenty fluid. My version is already much more fluid than occivink's due to being much simpler by using video-align, so I don't think more can be done, short of converting it all to native code, but I don't even know if that would help and it would be too complicated.

**Comment by zhouxinghong** [comment]
> QMPlay2 is more fluid but mpv's version is also plenty fluid. My version is already much more fluid than occivink's due to being much simpler by using video-align, so I don't think more can be done, short of converting it all to native code, but I don't even know if that would help and it would be too complicated.

I found such a great feature in QMPlay2 that I didn't hesitate to bring it fully into MPV, and it allows `drag-to-pan` videos to go beyond the window in QMPlay2 (which is great!). .

I'm totally in favor of converting all the code to native code to improve the smoothness, silkiness, naturalness, etc. when `drag-to-pan` videos.

Also, I'd like to add that during `drag-to-pan` videos, I need to press and hold the left mouse button very hard to make a `drag-to-pan` video, whereas in QMPlay2 you only need to press and hold the left mouse button very gently to make a `drag-to-pan` video (this is great!). So could this problem with `drag-to-pan` videos in MPV be considered for further improvement?

**Comment by guidocella** [comment]
You are welcome to implement any change you want. I will not change drag to pan further.

**Comment by zhouxinghong** [comment]
If I knew how to use C/C++/Lua then I wouldn't hesitate to come in and make the user experience of this feature the best it can be. 😜
